src/borgstore/backends/s3.py

```diff
@@ -0,0 +1,246 @@
+try:
+    import boto3
+except ImportError:
+    boto3 = None
+
+import re
+from typing import Optional
+
+from borgstore.constants import TMP_SUFFIX
+
+from ._base import BackendBase, ItemInfo, validate_name
+from .errors import BackendError, BackendMustBeOpen, BackendMustNotBeOpen, BackendDoesNotExist, BackendAlreadyExists
+from .errors import ObjectNotFound
+
+
+def get_s3_backend(url):
+    if boto3 is None:
+        return None
+
+    # s3:[profile|(access_key_id:access_key_secret)@][schema://hostname[:port]]/bucket/path
+    s3_regex = r"""
+        s3:
+        ((
+            (?P<profile>[^@:]+)  # profile (no colons allowed)
+            |
+            (?P<access_key_id>[^:@]+):(?P<access_key_secret>[^@]+)  # access key and secret
+        )@)?  # optional authentication
+        (?P<schema>[^:/]+)://
+        (?P<hostname>[^:/]+)
+        (:(?P<port>\d+))?/
+        (?P<bucket>[^/]+)/  # bucket name
+        (?P<path>.+)  # path
+    """
+    m = re.match(s3_regex, url, re.VERBOSE)
+    print(m)
```

Comment:
There shouldn't be any print() calls within the production code. Guess their output would "leak" into the output of e.g. borg when calling into borgstore.

Reply:
Agreed, will make the change.